bazel: upgrade rules_go to v0.32

rickystewart opened the pull request:
Also had to bump a couple deps (`go_googleapis`, `gazelle`) to match up
to what `rules_go` currently has at v0.32.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/82469)
<!-- Reviewable:end -->

rickystewart committed:
generate-distdir: include go_repository artifacts from `WORKSPACE`

Otherwise this stuff doesn't get included in the `distdir`.

Release note: None

rickystewart committed:
bazel: upgrade rules_go to v0.32

Also had to bump a couple deps (`go_googleapis`, `gazelle`) to match up
to what `rules_go` currently has at v0.32.

Release note: None

rail reviewed (approved):
Stamp.

rickystewart commented:
bors r=rail

rickystewart commented:
bors r-
